drivers/flash/flash_mcux_flexspi_nor.c

```diff
@@ -1008,6 +1008,37 @@ static int flash_flexspi_nor_check_jedec(struct flash_flexspi_nor_data *data,
 		/* Device uses bit 1 of status reg 2 for QE */
 		return flash_flexspi_nor_quad_enable(data, flexspi_lut,
 						     JESD216_DW15_QER_VAL_S2B1v5);
+	case 0x60ef:
```

Comment:
I have just noticed that device identification here is done by two bytes only, which is not correct.
The W25Q512 IQ, full id 0xef6020 or 0xef8020 (the second bytes indicate QE fixed/select),  has the same two bytes as w25Q128 0xef6019.
This means that the below identification may by mistake set 4 byte address config to 3 byte address device, which neither supports 133MHz operation nor configured commands.

Reply:
Good catch- would you like me to add a commit to this PR extending the device ID check for all the flash chips here?

Reply:
I think that this can be too much of a  change for this PR, just add if within case to filter out the w25Q128, using the last byte, or configure it properly. 
I do not think it is proper to fix everything in this PR and you will have to go through docs to gather info which will probably delay the PR till  you are ready.

Reply:
Ok, I've added a check to skip configuration if the full ID does not match the W25Q512NW. I don't have hardware to verify on at the moment, but reading the datasheet I believe the vendor ID should be read as `0x2060ef`